cli/command/cli.go

```diff
@@ -210,7 +210,13 @@ func (cli *DockerCli) Initialize(opts *cliflags.ClientOptions, ops ...Initialize
 	cli.configFile = cliconfig.LoadDefaultConfigFile(cli.err)
 
 	if cli.client == nil {
-		cli.contextStore = store.New(cliconfig.ContextStoreDir(), cli.contextStoreConfig)
+		baseContextSore := store.New(cliconfig.ContextStoreDir(), cli.contextStoreConfig)
+		cli.contextStore = &ContextStoreWithDefault{
```

Comment:
Should `ContextStoreWithDefault` be in its own package? So that we could do something like;

```go
storewithdefault.New(...)
```


Perhaps even integrate with the default `store` (but not sure if there's situations where we want that to be used separate)

Reply:
`ContextStoreWithDefault` needs things defined in package `command` (related to resolving the docker context based on "legacy" cli flags and env var). That would:
- either introduce a cycling dependency (dockerCli.Initialize needs this ContextStoreWithDefault)
- or move much of the command package code in a separate package
- implementing that in `store` package might introduce many new dependencies to the `store` package (and I don't feel like it is a good idea)

Reply:
Could we pass those options in the constructor?